src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -201,21 +215,74 @@ def process(
                     cc.message(grant_statements)
                     collected_output.append(grant_statements)
 
-                self.deannotate(py_file, extension_fns)
+                with open(sql_file, "a") as file:
+                    file.write("\n")
+                    file.write(create_stmt)
+                    if grant_statements is not None:
+                        file.write("\n")
+                        file.write(grant_statements)
+
+            self.deannotate(py_file, extension_fns)
+
+        if collected_sql_files:
+            edit_setup_script_with_exec_imm_sql(
+                collected_sql_files=collected_sql_files,
+                deploy_root=bundle_map.deploy_root(),
+                generated_root=resolved_generated_root,
+            )
 
         return "\n".join(collected_output)
 
-    def _normalize(self, extension_fn: NativeAppExtensionFunction, py_file: Path):
+    def _normalize_imports(
+        self,
+        extension_fn: NativeAppExtensionFunction,
+        py_file: Path,
+        deploy_root: Path,
+    ):
+        normalized_imports: Set[str] = set()
+        # Add the py_file, which is the source of the extension function
+        normalized_imports.add(f"/{to_stage_path(py_file)}")
+
+        for raw_import in extension_fn.imports:
+            if not Path(deploy_root, raw_import).exists():
+                # This should capture import_str of different forms: stagenames, malformed paths etc
+                # But this will also return True if import_str == "/". Regardless, we append it all to normalized_imports
+                cc.warning(
+                    f"{raw_import} does not exist in the deploy root. Skipping validation of this import."
+                )
+
+            if raw_import.startswith(STAGE_PREFIX) or raw_import.startswith("/"):
+                normalized_imports.add(raw_import)
+            else:
+                normalized_imports.add(f"/{to_stage_path(Path(raw_import))}")
+
+        # To ensure order when running tests
+        sorted_imports = list(normalized_imports)
+        sorted_imports.sort()
+        extension_fn.imports = sorted_imports
+
+    def _normalize(
+        self,
+        extension_fn: NativeAppExtensionFunction,
+        py_file: Path,
+        deploy_root: Path,
+    ):
         if extension_fn.name is None:
             # The extension function was not named explicitly, use the name of the Python function object as its name
             extension_fn.name = extension_fn.handler
 
         # Compute the fully qualified handler
+        # If user defined their udf as @udf(lambda: x, ...) then extension_fn.handler is <lambda>.
```

Comment:
These shouldn't even be present in the output from the collection subprocess, are you saying they're currently not filtered out?

Reply:
Yeah, they'll be there alright, and they're tricky to filter out because Python doesn't expose the fact that they're lambdas in a reliable way. `inspect` might give us clues, or we can filter out anything that has the same name as `lambda None` (just to make sure we're a bit resilient to changes in names, although that's pretty unlikely).